### ISSUE TITLE
Feature/Add data update automation

### ISSUE THREAD

**Issue by B13rg** [deadlock-wiki/deadbot#44]
This is an issue to capture the work to:

* create github action that checks for, generates the data, and creates a PR for the latest released version
* trigger automation through `workflow_dispatch`
* script that updates files based on [steam api](https://steamapi.xpaw.me/#IGCVersion_1422450) or seeing that files in an s3 bucket have changed
* investigate creating github releases for each game update release

**Comment by B13rg** [comment]
Example workflow for CS: https://github.com/ckreisl/cs-updates-as-json/blob/main/.github/workflows/poll-cs-updates.yml

**Comment by hassayag** [comment]
Also if its possible, a way to automate uploading to s3. Right now its just me remembering each patch, but that doesn't consider:
- Me missing each major patch
- The changes made in minor patches that don't come with a forum announcement

I have no idea how to do this, I know you can use steamdb to download older versions of a game, so maybe some routine that lets us download nad upload the last X patches that are missing

**Comment by hassayag** [comment]
> create github action that checks for, generates the data, and creates a PR for the latest released version

Oo I like! We should consider alternate routes for the data upload. I hate the massive diffs that we need to deal with. 

What are your thoughts on a separate repo to house the output data?

**Comment by B13rg** [comment]
Yeah, a separate repo might be the way to go.  then it might be easier to give each version it's own folder.
Looking at stuff,  the github runner could use https://github.com/Skippeh/steamfetch-action to download the game files.  
The credentials will be stored as repo secrets (read only).
It could be triggered by a manual_dispatch, and generate the output based on the version.
We could also limit who could run it, either to specific users or class of user.

I'm not sure how 2-factor auth will behave, unless there's some sort of steam-api key that can be generated/used instead of username+password+2fa

**Comment by hassayag** [comment]
Created a specific ticket for release process in #61 . feel free to flesh it out

**Comment by hassayag** [comment]
Similarly with #64

**Comment by B13rg** [comment]
Added a workflow to the `deadlock-data` repo that will sync itself off of this repo's data, it should hopefully be easy to build on!
You can manually trigger it, it'll sync against this repo: https://github.com/deadlock-wiki/deadlock-data/actions/workflows/scheduled-sync.yml

**Comment by B13rg** [comment]
Example run: https://github.com/deadlock-wiki/deadlock-data/commit/f657dee2b0e6fde65296e65b568c77a16c5d0967

**Comment by hassayag** [comment]
@B13rg is this issue still relevant? As there is crossover with other issues
